test/libsyncengine/update_detection/file_system_observer/testlocalfilesystemobserverworker.cpp

```diff
@@ -445,9 +441,25 @@ void TestLocalFileSystemObserverWorker::testLFSOFastMoveDeleteMove() { // MS Off
     CPPUNIT_ASSERT(_syncPal->snapshot(ReplicaSide::Local)->exists(std::to_string(fileStat.inode)));
 }
 
-bool MockLocalFileSystemObserverWorker::waitForUpdate(int64_t timeoutMs) const {
+void TestLocalFileSystemObserverWorker::testInvalidateCounter() {
+    _syncPal->_localFSObserverWorker->invalidateSnapshot();
+    CPPUNIT_ASSERT_EQUAL(true, _syncPal->snapshot(ReplicaSide::Local)->isValid()); // Snapshot is not invalidated yet
+    CPPUNIT_ASSERT_EQUAL(1, _syncPal->_localFSObserverWorker->_invalidateCounter);
+    _syncPal->_localFSObserverWorker->invalidateSnapshot();
+    CPPUNIT_ASSERT_EQUAL(true, _syncPal->snapshot(ReplicaSide::Local)->isValid()); // Snapshot is not invalidated yet
+    CPPUNIT_ASSERT_EQUAL(2, _syncPal->_localFSObserverWorker->_invalidateCounter);
+    _syncPal->_localFSObserverWorker->invalidateSnapshot();
+    CPPUNIT_ASSERT_EQUAL(false, _syncPal->snapshot(ReplicaSide::Local)->isValid()); // Snapshot is has been invalidated
+    CPPUNIT_ASSERT_EQUAL(0, _syncPal->_localFSObserverWorker->_invalidateCounter); // Counter has been reset
```

Comment:
Public accessor

Reply:
As mention above, will be tackle in a specific task.
However, I would argue in favor of `friend` classes to be used for testing purpose only. This would allow testing the values of certain variables (in this example `_invalidateCounter`) without exposing them publicly.

Reply:
An intermediary solution could be a `protected` accessor, which the existing `Mocked LFSO` could make public.

Reply:
Sure, but I would argue that creating an accessor, even a `protected` one, to be only used in test is perhaps not necessary and `friend` classes simplify the writing of code in this case.
But this is something we should discuss all together and agreed on. Then apply the chosen solution everywhere in code.

Reply:
Maybe we should not access the private members, the test can ensure that invalidateSnapshot work as expected without dealing with the internal logic of `LocalFileSystemObserverWorker`. 

```cpp
void TestLocalFileSystemObserverWorker::testInvalidateCounter() {
    _syncPal->_localFSObserverWorker->invalidateSnapshot();
    CPPUNIT_ASSERT_EQUAL(true, _syncPal->snapshot(ReplicaSide::Local)->isValid()); // Snapshot is not invalidated yet.
    
    _syncPal->_localFSObserverWorker->invalidateSnapshot();
    CPPUNIT_ASSERT_EQUAL(true, _syncPal->snapshot(ReplicaSide::Local)->isValid()); // Snapshot is not invalidated yet.
    
    _syncPal->_localFSObserverWorker->invalidateSnapshot();
    CPPUNIT_ASSERT_EQUAL(false, _syncPal->snapshot(ReplicaSide::Local)->isValid()); // Snapshot has been invalidated.

    Utility::msleep(1000); // Wait for the snapshot to be rebuilt

    CPPUNIT_ASSERT_EQUAL(true, _syncPal->snapshot(ReplicaSide::Local)->isValid()); // Snapshot is now valid again.
    
     _syncPal->_localFSObserverWorker->invalidateSnapshot();
    CPPUNIT_ASSERT_EQUAL(true, _syncPal->snapshot(ReplicaSide::Local)->isValid()); // Snapshot is not invalidated yet.
    
    _syncPal->_localFSObserverWorker->invalidateSnapshot();
    CPPUNIT_ASSERT_EQUAL(true, _syncPal->snapshot(ReplicaSide::Local)->isValid()); // Snapshot is not invalidated yet.
    
    _syncPal->_localFSObserverWorker->invalidateSnapshot();
    CPPUNIT_ASSERT_EQUAL(false, _syncPal->snapshot(ReplicaSide::Local)->isValid()); // Snapshot has been invalidated.

}
```

It is a bit longer but less dependant of the internal/private logic of `LocalFileSystemObserverWorker`

Reply:
Okay. It looks like people tends to agree to test only public method, not private one. Perhaps that is something that we should start applying.
Therefor, in this context, `_invalidateCounter` is an implementation detail that does not need to be accessed in the test.

Reply:
Done, as you suggested, in a21dc127b4413892df24be7947f11fa0c065bedf